src/client/datascience/interactive-window/nativeInteractiveWindowProvider.ts

```diff
@@ -209,6 +231,27 @@ export class NativeInteractiveWindowProvider implements IInteractiveWindowProvid
         });
     }
 
+    private update() {
+        const windows: NativeInteractiveWindow[] = [];
+
+        this._windows.forEach((win) => {
+            const notebookDocument = workspace.notebookDocuments.find(
+                (document) => win.notebookUri?.toString() === document.uri.toString()
```

Comment:
Can we not use the following
```suggestion
                (document) => win.notebookUri === document
```

Reply:
With the change I suggested we wontn' end up with situations where `notebookUri` is undefined.

Reply:
I'll suggest we either check `document` or `uri.toString()`.